packages/flutter_tools/test/general.shard/protocol_discovery_test.dart

```diff
@@ -112,6 +112,13 @@ void main() {
         expect(discoverer.uri, throwsA(isFormatException));
       });
 
+      testUsingContext('uri is null when the log reader closes early', () async {
+        initialize();
+        final Future<Uri> uriFuture = discoverer.uri;
+        await logReader.dispose();
+        expect(await uriFuture, isNull);
```

Comment:
nit: separate the expectation from the setup with a newline

Reply:
Done.